validator/node/node.go

```diff
@@ -582,38 +594,51 @@ func proposerSettings(cliCtx *cli.Context) (*validatorServiceConfig.ProposerSett
 			if err := warnNonChecksummedAddress(option.FeeRecipient); err != nil {
 				return nil, err
 			}
-			if option.BuilderConfig != nil {
-				option.BuilderConfig.GasLimit = reviewGasLimit(option.BuilderConfig.GasLimit)
+			if option.Builder != nil {
+				option.Builder.GasLimit = reviewGasLimit(option.Builder.GasLimit)
 			} else {
 				builderConfig, err := BuilderSettingsFromFlags(cliCtx)
 				if err != nil {
 					return nil, err
 				}
-				option.BuilderConfig = builderConfig
+				option.Builder = builderConfig.ToPayload()
 			}
-			vpSettings.ProposeConfig[bytesutil.ToBytes48(decodedKey)] = &validatorServiceConfig.ProposerOption{
+			o := &validatorServiceConfig.ProposerOption{
 				FeeRecipientConfig: &validatorServiceConfig.FeeRecipientConfig{
 					FeeRecipient: common.HexToAddress(option.FeeRecipient),
 				},
-				BuilderConfig: option.BuilderConfig,
+				BuilderConfig: validatorServiceConfig.ToBuilderConfig(option.Builder),
+			}
+			pubkeyB := bytesutil.ToBytes48(decodedKey)
+			vpSettings.ProposeConfig[pubkeyB] = o
+		}
+		if psExists {
```

Comment:
if no settings are saved don't you want to save them ? trying to understand the purpose of `psExists` here

Reply:
right so if the default settings are being updated witth `proposer-settings-file`,`--suggested-fee-recipient`,`--enable-builder` it will update the existing data, whereas if we are restarting with the proposer settings populating specific settings for validators it should just reset the whole thing, otherwise there's no way to clear the set values. We could also introduce a flag to do this if we want... 

if there is something that exists we save a new copy.

Reply:
So proposer settings will only exist in the db, if its a fresh run with the proposer-settings file ?

Reply:
proposer settings will load from db and not be overwritten if starting without proposer-settings-file. I'll write down all the cases in the description of the PR and transfer it to some docs page.